### PR TITLE
Hwkinvent-24 - port pinger to new inventory bus + setup work

### PR DESCRIPTION
**Pull request by Jiri-Kremser**
this is pr #72 + the code responsible for creating the tenant, environment and resourceType. It conceptually fits the best to the Pinger.

This can be merged after the PRs in hawkular/ui-{services|components} are merged

### CODE REVIEW

File: modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java [hunk 1]
```diff
@@ -65,31 +72,64 @@
...
-                    String type = (String) m.get("type");
-                    Map<String, String> params = (Map<String, String>) m.get("parameters");
-                    String url = params.get("url");
-                    destinations.add(new PingDestination(id, url));
+        int attempts = 0;
+        while (attempts++ < 10) {
+            try {
+                Client client = ClientBuilder.newClient();
+                // TODO: inventory does not have to be co-located
+                final String host = "http://localhost:8080";
```
Comment (pilhuhn) [inline modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java]: Pull out of loop? (+next line)

File: modules/pinger/src/main/java/org/hawkular/component/pinger/Log.java [hunk 2]
```diff
@@ -27,14 +28,14 @@
  * @author Heiko W. Rupp
  */
 @MessageLogger(projectCode = "HAWKULAR")
-public interface Log {
+public interface Log extends BasicLogger {
```
Comment (pilhuhn) [inline modules/pinger/src/main/java/org/hawkular/component/pinger/Log.java]: Just curious: why is that needed?

File: modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java [hunk 3]
```diff
@@ -53,7 +59,8 @@
     // How long do we wait between each round
     private static final int WAIT_MILLIS = 500;
 
-    String tenantId = "test";
+    private final String tenantId = "test";
```
Comment (pilhuhn) [inline modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java]: Shouldn't those be retrieved from Inventory now that Inventory supports them?

Comment (Jiri-Kremser) [inline modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java]: someone has the create the default tenant and environment. This should be probably done by the keycloak ui code, this is just to make it working w/ the 'end-to-end' test

File: modules/pinger/src/main/java/org/hawkular/component/pinger/Log.java (hunk 2 above)
Comment (Jiri-Kremser) [inline modules/pinger/src/main/java/org/hawkular/component/pinger/Log.java]: It isn't (http://docs.jboss.org/seam/3/3.1.0.Final/reference/en-US/html/solder-logging.html#jbosslogging), but if you do that, you'll get the generic methods like debug, trace, logf, etc.

If we want to be super strict and allow only the predefined (and localized) methods than it can go away. Btw. alert does it the same way.

File: modules/pinger/pom.xml [hunk 4]
```diff
@@ -47,7 +47,13 @@
     <dependency>
       <groupId>org.hawkular.inventory</groupId>
       <artifactId>inventory-api</artifactId>
-      <version>1.0.0-SNAPSHOT</version>
+      <version>0.0.1-SNAPSHOT</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.hawkular.inventory</groupId>
+      <artifactId>bus-api</artifactId>
```
Comment (metlos) [inline modules/pinger/pom.xml]: This needs to be changed to "inventory-bus-api" now that https://github.com/hawkular/hawkular-inventory/pull/49 is merged.

Comment (Jiri-Kremser) [inline modules/pinger/pom.xml]: done
